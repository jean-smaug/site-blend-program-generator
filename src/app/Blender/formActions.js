/**
 * Add a Keyword to the state
 * @param word
 * @returns {{type: string, data: {word: *}}}
 */
export const addKeyword = word => ({
  type: 'ADD_KEYWORD',
  data: {
    word,
  },
});

/**
 * Remove a Keyword to the state
 * @param word
 * @returns {{type: string, data: {word: *}}}
 */
export const removeKeyword = word => ({
  type: 'REMOVE_KEYWORD',
  data: {
    word,
  },
});

/**
 * Add a Domain to the state
 * @param domain
 */
export const addDomain = domain => ({
  type: 'ADD_DOMAIN',
  data: {
    domain,
  },
});

/**
 * Remove a Domain to the state
 * @param domain
 */
export const removeDomain = domain => ({
  type: 'REMOVE_DOMAIN',
  data: {
    domain,
  },
});

/**
 * Update level of Domain to the state
 * @param domain
 */
export const updateLevel = domain => ({
  type: 'UPDATE_LEVEL_DOMAIN',
  data: {
    domain,
  },
});

/**
 * Add an objectif to the state
 * @param objectif
 * @returns {{type: string, data: {objectif: *}}}
 */
export const addObjectif = objectif => ({
  type: 'ADD_OBJECTIF',
  data: {
    objectif,
  },
});

/**
 * Remove an objectif from the state
 * @param objectif
 * @returns {{type: string, data: {objectif: *}}}
 */
export const removeObjectif = objectif => ({
  type: 'REMOVE_OBJECTIF',
  data: {
    objectif,
  },
});
