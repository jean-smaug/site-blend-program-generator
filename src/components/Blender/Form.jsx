import React from 'react';
import * as keywords from './data/keywords.json';
import * as thematiques from './data/thematiques.json';
import * as objectifs from './data/objectifs.json';
import CheckboxKeyword from './CheckboxKeyword/CheckboxKeyword';
import CheckboxDomain from './CheckboxDomain/CheckboxDomain';
import CheckboxObjectif from './CheckboxObjectif/CheckboxObjectif';
import Mixeur from './Mixeur/Mixeur';
import './Form.css';

const Form = () => {
  const showKeyword = keywords.map(item => (
    <CheckboxKeyword item={item} key={item.id} />
  ));
  const showTheme = thematiques.map(item => (
    <CheckboxDomain item={item} key={item.id} />
  ));
  const showObjectif = objectifs.map(item => (
    <CheckboxObjectif item={item} key={item.id} />
  ));
  return (
    <div className="form">
      <div className="items">
        <div className="item">
          <h1 className="category-title">Les th√©matiques</h1>
          { showTheme }
        </div>
        <div className="item">
          <h1 className="category-title">Les sujets</h1>
          { showKeyword }
        </div>
        <div className="item">
          <h1 className="category-title">Vos objectifs</h1>
          { showObjectif }
        </div>
        <div className="item">
          <Mixeur />
        </div>
      </div>
    </div>
  );
};

export default Form;
