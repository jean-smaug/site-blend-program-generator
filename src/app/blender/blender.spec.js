import React from 'react';
import { shallow } from 'enzyme';
import Blender from './blender.component';

describe('blender.component', () => {
  it('should render blender component', () => {
    const wrapper = shallow(<Blender />);
    expect(wrapper.getNodes()).toMatchSnapshot();
  });
});
