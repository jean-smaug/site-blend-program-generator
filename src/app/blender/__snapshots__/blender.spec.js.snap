// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blender.component should render blender component 1`] = `
Array [
  <div>
    <div
        className="form"
    >
        <div
            className="form-header"
        >
            <h1>
                Choisissez les ingrédients de vos smoothies
            </h1>
            <h2>
                et laissez-nous vous proposer un BlendWebMix sur mesure...
            </h2>
        </div>
        <div
            className="columns"
        >
            <div
                className="modal-wrap column"
            >
                <div
                    className="modal-header"
                >
                    <span
                        className="is-active"
                    />
                    <span
                        className={null}
                    />
                    <span
                        className={null}
                    />
                </div>
                <div
                    className="modal-bodies"
                >
                    <div
                        className="modal-body"
                    >
                        <div
                            className="items"
                        >
                            <div
                                className="columns"
                            >
                                <div
                                    className="column"
                                >
                                    <h1
                                        className="category-title"
                                    >
                                        Vos sujets favoris
                                    </h1>
                                    <h2
                                        className="category-desc"
                                    >
                                        Choisissez vos types de conférences préférées ainsi que le niveau souhaité.. :)
                                    </h2>
                                    <hr />
                                    <div
                                        className="columns"
                                    >
                                        <div
                                            className="column"
                                        >
                                            <Connect(CheckboxDomainComponent)
                                                item={
                                                    Object {
                                                        "id": "tech",
                                                        "libelle": "Technique",
                                                      }
                                                }
                                            />
                                        </div>
                                        <div
                                            className="column"
                                        >
                                            <Connect(CheckboxDomainComponent)
                                                item={
                                                    Object {
                                                        "id": "market",
                                                        "libelle": "Market",
                                                      }
                                                }
                                            />
                                        </div>
                                        <div
                                            className="column"
                                        >
                                            <Connect(CheckboxDomainComponent)
                                                item={
                                                    Object {
                                                        "id": "biz",
                                                        "libelle": "Business",
                                                      }
                                                }
                                            />
                                        </div>
                                        <div
                                            className="column"
                                        >
                                            <Connect(CheckboxDomainComponent)
                                                item={
                                                    Array [
                                                        Object {
                                                          "id": "tech",
                                                          "libelle": "Technique",
                                                        },
                                                        Object {
                                                          "id": "market",
                                                          "libelle": "Market",
                                                        },
                                                        Object {
                                                          "id": "biz",
                                                          "libelle": "Business",
                                                        },
                                                      ]
                                                }
                                            />
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>
                        <hr />
                        <div
                            className=" columns"
                        >
                            <div
                                className="column is-4"
                            />
                            <div
                                className="column is-4 is-offset-6"
                            >
                                <input
                                    className="btn-suivant"
                                    onClick={[Function]}
                                    type="button"
                                    value="Suivant >"
                                />
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>
</div>,
]
`;
